### PR TITLE
AI-170: Added websocket_input and websocket_output components

### PR DESCRIPTION
**Pull request by efunneko**
## Main changes

Added two new components:

    * websocket_input
    * websocket_output

These components allow flow input to come from a websocket connection and will can send flow data to a websocket connection. They both support the same format and encoding as the broker input/output components support.

I have also added a small webapp in examples/websocket and config file to provide a simple example of how this can work. The webapp allows you to connect to the websocket server in the component, push in some JSON and it will be bounced back to you on that same websocket connection and displayed on the page.

websocket_input supports serving a single page specified in the component config specifically allow one webapp to work with it (as in the example).

## Primary reason for this component

This component has been added as an alternative mechanism to get events from the broker to a webpage. For our purposes, this will allow our visualizer app to be independent of of the broker type so that we can have a dumb dev broker and not lose the visualize functionality.

### CODE REVIEW

File: src/solace_ai_connector/components/inputs_outputs/websocket_base.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        )
+        self.setup_websocket()
+
+        if self.serve_html:
+            self.setup_html_route()
+
+    def setup_html_route(self):
+        @self.app.route("/")
+        def serve_html():
+            return send_file(self.html_path)
```
Comment (cyrus2281) [inline src/solace_ai_connector/components/inputs_outputs/websocket_base.py:88]: Wouldn't serve directory be bettor? to support css, js, and asset files?

File: examples/websocket/websocket_example_app.html [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+         const jsonInput = document.getElementById('json-input');
+         const sendButton = document.getElementById('send-button');
+         const inputError = document.getElementById('input-error');
+         const messagesDiv = document.getElementById('messages');
+         const clearButton = document.getElementById('clear-button');
+
+         let socket;
+
+         function connect() {
+             socket = io(urlInput.value);
```
Comment (cyrus2281) [inline examples/websocket/websocket_example_app.html:194]: Aren't the input and output endpoints on 2 different servers?

Comment (efunneko) [reply examples/websocket/websocket_example_app.html:194]: No. By default, the websocket_output will look into the global pool of open websocket connections using the socket_id from its input. It is possible, but not typical, for the websocket_output to have a server, but that should only be for cases where it isn't a request/response type of application.

File: src/solace_ai_connector/components/inputs_outputs/websocket_base.py (hunk 1 above)
Comment (efunneko) [reply src/solace_ai_connector/components/inputs_outputs/websocket_base.py:88]: I considered this and I could be wrong with my choice. I didn't think it really made sense to use this as a general webserver, but I wanted a really simple demo that wouldn't require you to create a webserver... So I just served a single file. By doing this, you don't need to know the name of the file being served.

**Pull request merged**